src/modules/CRUD/services/CRUDProvider.js

```diff
@@ -29,105 +33,185 @@ export class CRUDProvider extends React.Component {
   setMap = map => !this.isUnmount && this.setState({ map });
 
   getMapConfig = async () => {
+    const result = {};
     try {
-      const { results: mapConfig = {} } = await fetchMapConfig();
-      this.setState({ mapConfig });
+      const mapConfig = await fetchMapConfig();
+      result.mapConfig = mapConfig;
     } catch (e) {
-      this.setState(state => ({
-        ...state,
-        errors: { ...state.errors, code: e.message },
-      }));
+      result.error = e;
     }
+
+    const { mapConfig = {}, error } = result;
+    this.setState(({ errors }) => ({
+      mapConfig,
+      errors: {
+        ...errors,
+        mapConfig: error,
+      },
+    }));
+
+    return mapConfig;
   };
 
   getSettings = async () => {
+    const result = {};
     try {
       const settings = await fetchSettings();
-      this.setState({ settings });
+      result.settings = settings;
     } catch (e) {
-      this.setState(state => ({
-        ...state,
-        errors: { ...state.errors, code: e.message },
-      }));
+      result.error = e;
+    }
+
+    const { settings = {}, error } = result;
+
+    this.setState(({ errors }) => ({
+      settings,
+      errors: {
+        ...errors,
+        settings: error,
+      },
+    }));
+
+    return settings;
+  }
+
+  getFormattedError = props => {
+    const {
+      store,
+      error,
+      ids,
+    } = props;
+    const { errors: { [store]: errorStore } } = this.state;
+
+    const propsFromIds = Object.keys(ids);
+
+    // Remove error if it was before in the store
+    if (!(error instanceof Error)) {
+      return errorStore.filter(item => (
```

Comment:
What kind of *item* are stored in `errorStore`? Wouldn't, for example, `error` be more meaningful than `item`?